fix: align Duration picker layout and background color in child tables

manzoorsofi opened the pull request:
This PR fixes a UI inconsistency in the Duration field inside child tables.
The Duration picker (Days, Hours, Minutes, Seconds) in child tables had the following issues:

Missing background color on numeric inputs

Misaligned layout, with the "Seconds" input wrapping to a new line

These issues did not appear in the parent form Duration field.
Attached screenshots show both the broken and corrected layouts.
<img width="1047" height="409" alt="Screenshot 2025-11-11 110616" src="https://github.com/user-attachments/assets/70094f07-5d8f-42f7-b9bc-e8cdca64145f" />
<img width="1769" height="783" alt="Screenshot 2025-11-11 115053" src="https://github.com/user-attachments/assets/ad9b192a-ec8f-45fa-8716-66f311074229" />

manzoorsofi committed:
fix: align Duration picker layout and background color in child tables

barredterra commented:
From your screenshot, it looks like the duration picker is still "inside" the table and doesn't pop "over" it. I think we may be able to solve this by using `position: fixed`.

manzoorsofi committed:
fix: align Duration picker layout and background color in child tables

manzoorsofi commented:
Hi @barredterra 
This update refines the earlier fix for the Duration field UI in child tables.

Replaced var(--control-bg) with var(--gray-100) for the duration input background color,
since --control-bg is not inherited properly within the child table (grid) context.

Ensured consistent background color for duration inputs across both parent and child forms.

Verified that the layout and alignment of the “Seconds” input remain correct and positioned properly.

Addressed SCSS formatting issues as per pre-commit hook requirements.
<img width="1713" height="756" alt="Screenshot 2025-11-11 165806" src="https://github.com/user-attachments/assets/bb3cb0ce-20e3-48be-aef2-fdcce02aa821" />

barredterra commented:
@manzoorsofi from your screenshot, the two pickers still look very different. The one in the table has more vertical padding, vertical lines and no dark input background.

A note regarding commit hygiene: `style:` commits are supposed to contain only changes that don't affect code meaning (e.g., formatting, whitespace, semicolons). [669ae3f](https://github.com/frappe/frappe/pull/34661/commits/669ae3f571bfb2df20f476aa0386192cda432308) also contains all the requested fixes, which is unexpected.

manzoorsofi commented:
@barredterra I realized I hadn’t uploaded the updated screenshot earlier — please check the last comment now.
I haven’t changed or added any whitespace or semicolons.
The only change made is replacing var(--control-bg) with var(--gray-100), since --control-bg isn’t inherited correctly within the child table (grid) context.
Both var(--control-bg) and var(--gray-100) represent the same color (#f3f3f3).
Also, the child view already has vertical lines by default.

barredterra commented:
> I haven’t changed or added any whitespace or semicolons.

Exactly, so the commit type should not have been `style: `.

> Also, the child view already has vertical lines by default.

I think the vertical bars being present in the child table duration picker is part of the broken layout we're trying to fix here.

manzoorsofi commented:
@barredterra 
I’ve removed the unnecessary vertical grid lines that were appearing in the Duration picker inside child tables.

The layout is now aligned properly with the parent field, and the background color remains consistent.

Let me know if any further UI fine-tuning or cleanup is needed.

<img width="1579" height="815" alt="Screenshot 2025-11-11 215453" src="https://github.com/user-attachments/assets/eed8d494-e440-47d7-a7d3-ab80ad3d4dfd" />

barredterra committed:
fix: padding, move duration styling to controls.scss

barredterra committed:
fix: background color that works in dark mode

barredterra commented on frappe/public/scss/common/form.scss:
```diff
@@ -28,3 +29,9 @@
+	font-size: var(--text-xl);
+	font-weight: 700;
+	color: var(--heading-color);
+}
```
@manzoorsofi how are these changes related?

manzoorsofi replied:
@barredterra I checked,— it seems the .head-title block got duplicated by mistake. Should I remove it and commit the fix again?

barredterra replied:
Just add a new commit removing the duplicate code. 👍🏼

manzoorsofi committed:
chore: remove duplicate .head-title definition

manzoorsofi commented:
@barredterra Done 👍🏼
